src/main/java/homework/seven/ArrayValueCalculator.java

```diff
@@ -0,0 +1,30 @@
+package homework.seven;
+
+public class ArrayValueCalculator {
+    String[][] array = new String[4][4];
+
+    public int doCalc(String[][] array) throws ArraySizeException, ArrayDataException {
+        int suma = 0;
+        this.array = array;
+        if (this.array.length > 4) {
+            throw new ArraySizeException();
+        } else {
+            for (int i = 0; i < 4; i++) {
+                if (this.array[i].length > 4) {
+                    throw new ArraySizeException("index "+i);
+                } else {
+                    for (int j = 0; j < 4; j++) {
+                        System.out.print(array[i][j] + " ");
+                        try {
+                            suma += Integer.parseInt(array[i][j]);
+                        } catch (NumberFormatException e) {
+                            throw new ArrayDataException(i, j, array[i][j]);
+                        }
+                    }
+                }
+                System.out.println("> " + suma);
+            }
+        }
```

Comment:
In your solution possible to throw ArrayDataException for an invalid array size (ArraySizeException should be thrown because the array size is invalid, but we check before cells format). Example
String[][] arr = {
{"1", "XXX", "567", "1234"},
{"45"},
{"45", "998", "567", "1234"},
{"45", "998", "567", "456"}
};

Reply:
first.{"1", "XXX", "567", "1234"} will be error {Cannot convert character [ XXX ] to number  of index number [0 ] [ 1 ]}
i fixed {"1", "33", "567", "1234"}
second.{"45"}, will be Array size is index 1 = 1 should be 4 
i fixed